src/riak_ttb_codec.erl

```diff
@@ -58,21 +58,15 @@ return_resp({Atom, <<>>}) ->
 return_resp(Resp) ->
     Resp.
 
-de_stringify(Tuple) when is_tuple(Tuple) ->
-    list_to_tuple(de_stringify(tuple_to_list(Tuple)));
-de_stringify(List) when is_list(List), is_integer(hd(List)) ->
-    %% Yes, this could corrupt utf-8 data, but we should never, ever
-    %% have put it in string format to begin with
-    list_to_binary(List);
-de_stringify(List) when is_list(List) ->
-    lists:map(fun de_stringify/1, List);
-de_stringify(Element) ->
-    Element.
-
 encode_ts_rows(Rows) ->
     [encode_ts_row(Row) || Row <- Rows].
 
-encode_ts_row(Row) when is_list(Row) ->
-    list_to_tuple(Row);
 encode_ts_row(Row) when is_tuple(Row) ->
-    Row.
+    tuple_to_list(Row);
```

Comment:
When will `Row` be a tuple, if ever? If `Row` should *never* be a tuple, this function clause shouldn't exist so that a crash will happen.

If `Row` is allowed to be a tuple, should the code do this instead to ensure `undefined` is converted to `[]` ?

```erlang
encode_ts_row(Row) when is_tuple(Row) ->
    encode_ts_row(tuple_to_list(Row));
```